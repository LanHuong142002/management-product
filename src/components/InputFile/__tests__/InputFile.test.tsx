import { act, cleanup, fireEvent, render, screen } from '@testing-library/react';
import { InputFile } from '@components';

describe('Testing input file component', () => {
  const handleChange = jest.fn();

  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    url: 'image.jpg',
    name: 'productImage',
    id: 'productImage',
    text: 'Upload Image',
  };

  it('Should render Input File component', () => {
    const { container } = render(<InputFile {...defaultProps} onChange={handleChange} />);

    expect(container).toMatchSnapshot();
  });

  it('Should call onChange when click input file component', () => {
    const file = new File(['image.jpg'], 'image.jpg', { type: 'image/png' });
    const handleChange = jest.fn();
    render(<InputFile {...defaultProps} onChange={handleChange} />);

    const inputFile = screen.getByTestId('input-file-label');
    const input = inputFile.querySelector('input[type="file"]')!;

    act(() => {
      fireEvent.change(input, { target: { files: [file] } });
    });

    expect(handleChange).toBeCalled();
  });
});
