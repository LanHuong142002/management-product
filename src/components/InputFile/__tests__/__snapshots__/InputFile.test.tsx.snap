// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing input file component Should render Input File component 1`] = `
<div>
  <label
    class="input-file-wrapper input-file-primary"
    data-testid="input-file-label"
    for="productImage"
  >
    <figure
      class="image-wrapper image-size-xs "
    >
      <img
        alt="icon upload"
        class="image "
        src="image.jpg"
      />
    </figure>
    <div>
      Upload Image
      <input
        accept="image/png, image/jpeg"
        id="productImage"
        name="productImage"
        type="file"
      />
    </div>
  </label>
</div>
`;
